File: src/tsBusinessLayer/dtos/OrderDTO.ts
import {IOrder} from "../interfaces/IOrder"
import { OrderInvoiceDTO } from "./OrderInvoiceDTO";
import { ShippingMethodDTO } from "./ShippingMethodDTO";
import { UserDTO } from "./UserDTO";

export class OrderDTO implements IOrder{
    public id:number;
    public Customer: UserDTO;
    public ShippingMethod:ShippingMethodDTO;
    public OrderInvoices:Array<OrderInvoiceDTO>;
    public ShippingAddress?:string;
    public CreatedAt?: Date;
    public ShippingMethodPrice?: number;

    constructor(id:number, customer: UserDTO, shippingMethod: ShippingMethodDTO, orderInvoices:Array<OrderInvoiceDTO>, ShippingAddress?:string, CreatedAt?: Date, ShippingMethodPrice?:number){
        this.id = id;
        this.Customer = customer;
        this.ShippingMethod = shippingMethod;
        this.OrderInvoices = orderInvoices;
        if(ShippingAddress){
            this.ShippingAddress = ShippingAddress;
        }
        if(CreatedAt){
            this.CreatedAt = CreatedAt;
        }
        if(ShippingMethodPrice){
            this.ShippingMethodPrice = ShippingMethodPrice;
        }
    }
}

File: Berix-Library-API/Application/DTOs/Orders/OrderDTO.cs
using Application.DTOs.Books;
using Application.DTOs.ShippingMethods;
using Application.DTOs.Users;
using Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.DTOs.Orders
{
    public class OrderDTO
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public int ShippingMethodId { get; set; }
        virtual public ICollection<OrderInvoiceDTO> OrderInvoices { get; set; } = new List<OrderInvoiceDTO>();
        virtual public ShippingMethodDTO ShippingMethod { get; set; }
        virtual public UserDTO Customer { get; set; }
    }
}
